kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/formats/CsvReaderIteratorAdaptor.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 Lenses.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.lenses.streamreactor.connect.aws.s3.formats
+
+import au.com.bytecode.opencsv.CSVReader
+
+class CsvReaderIteratorAdaptor(csvReader: CSVReader) extends Iterator[Array[String]] {
+
+  private var currentRecord: Option[Array[String]] = None
+  private var nextRecord: Option[Array[String]] = Option(csvReader.readNext())
+
+  override def hasNext: Boolean = {
+    nextRecord.nonEmpty
+  }
+
+  def peek: Option[Array[String]] = {
+    hasNext
```

Comment:
this call is redundant

Reply:
how so?  I believe it is used so that it can peek at the next record here:
https://github.com/lensesio/stream-reactor/blob/f2dc893288a8b5c28bf6fd06fca0858aba9ab4c8/kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/formats/CsvFormatStreamReader.scala#L45

Reply:
hasNext calls nextRecord.nonEmpty = it does nothing. 
nextRecords is a var not a function.

Reply:
I misunderstood the comment and thought perhaps the method wasn't necessary

I've removed the unnecessary evaluation of hasNext